bots/github/cache.py

```diff
@@ -48,10 +48,13 @@ def prune(self):
         if not os.path.exists(self.directory):
             return
         now = time.time()
-        for filename in os.listdir(self.directory):
-            path = os.path.join(self.directory, filename)
-            if os.path.isfile(path) and os.stat(path).st_mtime < now - 7 * 86400:
-                os.remove(path)
+        try:
+            for filename in os.listdir(self.directory):
+                path = os.path.join(self.directory, filename)
+                if os.path.isfile(path) and os.stat(path).st_mtime < now - 7 * 86400:
+                    os.remove(path)
+        except OSError:
+            pass
```

Comment:
The intent is to catch OSError errors in listdir, isfile, stat, and remove. It seems that any of these can race when multiple bots are operating on the same cache. Should I make this specific to ENOENT and ENOTDIR?